src/pcre2_compile.c

```diff
@@ -7999,9 +8000,7 @@ for (;; pptr++)
     default:
     if (meta >= META_END)
       {
-#ifdef DEBUG_SHOW_PARSED
-      fprintf(stderr, "** Unrecognized parsed pattern item 0x%.8x\n", *pptr);
-#endif
```

Comment:
I replaced it with an UNREACHABLE assertion. The printf is currently disabled, so if the condition is hit, you won't see the logging. The assertion is nice and noisy, so will help development.

Reply:
if by disabled you mean that there is a commented out:

```c
#define DEBUG_SHOW_PARSED 1
```

then that is by design, as those are usually enabled on demand when additional debugging is needed

I Presume that Philip (or wohever uses that) would rather have a printf than an abort though.

Reply:
DEBUG_SHOW_PARSED is an ancient debugging feature that I put in when I was adding the initial parsing pre-pass for 10.23 and I wanted to see what its output was. I didn't think it would be of general use afterwards which is why I didn't imiplement anything more elaborate.

Reply:
I did some updates there but never actually tried it. Is there anybody using it?